### PR TITLE
ENT - RMP-475 "Quick search candidates MUST reset left filters by stage and the detailed field filters such that the search is applied to ALL candidates"

### PR DESCRIPTION
**Pull request by gayan-orangehrm**
Vue app fixes related to this defect is below,

**Before PR review**
https://websvn.orangehrm.com/orangehrm?op=comp&compare[]=/enterprise/branch/recruitment-vue-conversion/@170737&compare[]=/enterprise/branch/recruitment-vue-conversion/@170738

**After PR review**
https://websvn.orangehrm.com/orangehrm?op=comp&compare[]=/enterprise/branch/recruitment-vue-conversion/symfony/web/vue-app/src/@170783&compare[]=/enterprise/branch/recruitment-vue-conversion/symfony/web/vue-app/src/@170784
https://websvn.orangehrm.com/orangehrm?op=comp&compare[]=/enterprise/branch/recruitment-vue-conversion/symfony/web/vue-app/src/@170784&compare[]=/enterprise/branch/recruitment-vue-conversion/symfony/web/vue-app/src/@170785

Below files shows changes because I ran "yarn lint" locally

components/src/core/components/Input/Autocomplete/AutocompleteInput.vue
components/src/core/components/Input/DateInput.vue
components/src/core/components/Input/DropdownInput.vue
components/src/core/components/Input/MultiSelect/MultiSelectInput.vue
components/src/core/components/Input/Select/SelectInput.vue
components/src/core/components/Input/Time/TimeInput.vue
components/src/core/components/Pagination/Pagination.vue
components/src/directives/dropdown-direction/index.ts

### CODE REVIEW

File: components/src/core/components/TableSidebar/TableSidebar.vue [hunk 1]
```diff
@@ -135,18 +135,19 @@ export default defineComponent({
...
+    const selectedListItem = computed({
+      get() {
+        return {
+          id: props.selectedListItemId,
+          label: null,
+          active: false,
+        };
+      },
+      // eslint-disable-next-line
+      set() {},
```
Comment (Super-Chama) [inline components/src/core/components/TableSidebar/TableSidebar.vue:147]: remove setter if not needed.

```
const selectedListItem = computed(() => ({
     id: props.selectedListItemId,
     label: null,
     active: false,
}));
```

File: components/src/core/components/Pagination/Pagination.vue [hunk 2]
```diff
@@ -162,11 +162,11 @@ export default defineComponent({
     },
 
     showPagination() {
-      if ((this.totalRecordsCount <= this.pagesList[0]) && (this.length <= 1)) {
+      if (this.totalRecordsCount <= this.pagesList[0] && this.length <= 1) {
         return false;
       }
       return true;
-    }
+    },
```
Comment (Super-Chama) [inline components/src/core/components/Pagination/Pagination.vue:169]: is this correct?

Comment (Super-Chama) [reply components/src/core/components/Pagination/Pagination.vue:169]: never-mind i was confused as this curly brace is for  the if block

File: components/src/core/components/TableSidebar/TableSidebar.vue (hunk 1 above)
Comment (gayan-orangehrm) [reply components/src/core/components/TableSidebar/TableSidebar.vue]: Fixed

File: components/src/core/components/Pagination/Pagination.vue (hunk 2 above)
Comment (gayan-orangehrm) [reply components/src/core/components/Pagination/Pagination.vue:169]: Noted

File: components/src/core/components/TableSidebar/TableSidebar.vue [hunk 3]
```diff
@@ -75,7 +75,7 @@
 </template>
 
 <script lang="ts">
-import {defineComponent, computed, ref} from 'vue';
+import {defineComponent, computed, ref, toRef, Ref} from 'vue';
```
Comment (Super-Chama) [inline components/src/core/components/TableSidebar/TableSidebar.vue:78]: `toRef, Ref, ref` remove unused

**Review by Super-Chama — APPROVED** [review]

**Pull request merged**